### PR TITLE
feat: implement callbacks config getter/setter

### PR DESCRIPTION
**Pull request by ansiz**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/filter/stream/flowcontrol/callbacks.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ type Callbacks interface {
 	Append(ctx context.Context, headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap)
 	Exit(filter *StreamFilter)
 	Enabled() bool
+	SetConfig(conf *Config)
+	GetConfig() Config
```
Comment (antJack) [inline pkg/filter/stream/flowcontrol/callbacks.go:31]: 为什么 Set 的是指针，Get 返回的是拷贝？

File: pkg/filter/stream/flowcontrol/callbacks.go [hunk 2]
```diff
@@ -102,3 +106,13 @@ func (dc *DefaultCallbacks) Enabled() bool { return dc.config.GlobalSwitch }
...
 }
+
+// SetConfig sets the config of callbacks.
+func (dc *DefaultCallbacks) SetConfig(conf *Config) {
+	dc.config = conf
+}
+
+// GetConfig gets the readonly config of callbacks.
+func (dc *DefaultCallbacks) GetConfig() Config {
+	return *dc.config
```
Comment (antJack) [inline pkg/filter/stream/flowcontrol/callbacks.go:117]: 简单解引用并不能保证只读吧，Config.Rules 数组引用的还是同一个

File: pkg/filter/stream/flowcontrol/callbacks.go [hunk 3]
```diff
@@ -51,7 +53,9 @@ func GetCallbacksByConfig(conf *Config) Callbacks {
 	if !ok {
 		return &DefaultCallbacks{config: conf}
 	}
-	return cb.(Callbacks)
+	customizedCallbacks := cb.(Callbacks)
+	customizedCallbacks.SetConfig(conf)
```
Comment (antJack) [inline pkg/filter/stream/flowcontrol/callbacks.go:57]: check nil，RegisterCallbacks() 塞进去的可能是个空值

**Review by antJack — APPROVED** [review]
LGTM

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Pull request merged**